### PR TITLE
Marshal and SafeBuffer doc updates

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
## Summary

Update docs for `Marshal` and `SafeBuffer` classes.

Fixes https://github.com/dotnet/dotnet-api-docs/issues/3928

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
@GrabYourPitchforks FYI, I updated some of the UTF8 related APIs on the `Marshal` class.

File: xml/System.Runtime.InteropServices/Marshal.xml [hunk 1]
```diff
@@ -9477,10 +9493,22 @@ There is a difference in the behavior of the `GetLastWin32Error` method on .NET
         <Parameter Name="s" Type="System.String" Index="0" FrameworkAlternate="netcore-1.1;netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netstandard-2.1;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="s">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="s">A managed string to be copied.</param>
+        <summary>Copies the contents of a managed <see cref="T:System.String" /> to a block of memory allocated from the unmanaged COM task allocator.</summary>
+        <returns>An integer representing a pointer to the block of memory allocated for the string, or 0 if s is <see langword="null" />.</returns>
```
Comment (GrabYourPitchforks) [inline xml/System.Runtime.InteropServices/Marshal.xml]: `s` should be `<param ref="s">`?

File: xml/System.Runtime.InteropServices/Marshal.xml [hunk 2]
```diff
@@ -9477,10 +9493,22 @@ There is a difference in the behavior of the `GetLastWin32Error` method on .NET
...
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+ <xref:System.Runtime.InteropServices.Marshal.StringToCoTaskMemUTF8%2A> is useful for custom marshaling or for use when mixing managed and unmanaged code. Because this method allocates the unmanaged memory required for a string, always free the memory by calling <xref:System.Runtime.InteropServices.Marshal.FreeCoTaskMem%2A?displayProperty=nameWithType>. This method provides the opposite functionality of <xref:System.Runtime.InteropServices.Marshal.PtrToStringUTF8%2A?displayProperty=nameWithType>. The characters of the string are copied as UTF8 characters.
```
Comment (GrabYourPitchforks) [inline xml/System.Runtime.InteropServices/Marshal.xml]: May want to comment that we allocate enough room for the null terminator and stick it at the end of the buffer.

File: xml/System.Runtime.InteropServices/SafeBuffer.xml [hunk 3]
```diff
@@ -447,9 +447,19 @@ finally {
...
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+ The common language runtime uses this property to determine whether critical finalization is required.
+
+ ]]></format>
+        </remarks>
+        <altmember cref="P:System.Runtime.InteropServices.SafeHandleMinusOneIsInvalid.IsInvalid" />
+        <altmember cref="T:System.Runtime.InteropServices.SafeHandleMinusOneIsInvalid" />
```
Comment (carlossanlop) [inline xml/System.Runtime.InteropServices/SafeBuffer.xml]: These two lines threw a build warning:

```
2 invalid cross reference(s)
"<xref:System.Runtime.InteropServices.SafeHandleMinusOneIsInvalid.IsInvalid>",
"<xref:System.Runtime.InteropServices.SafeHandleMinusOneIsInvalid>".
```
I didn't find those two APIs inside the `System.Runtime.InteropServices` namespace. They live inside `Microsoft.Win32`:

```suggestion
        <altmember cref="P:Microsoft.Win32.SafeHandles.SafeHandleMinusOneIsInvalid.IsInvalid" />
        <altmember cref="T:Microsoft.Win32.SafeHandles.SafeHandleMinusOneIsInvalid" />
```

Comment (carlossanlop) [reply xml/System.Runtime.InteropServices/SafeBuffer.xml]: BTW How cool is it that we now have multi line suggestions? 😄

File: xml/System.Runtime.InteropServices/Marshal.xml [hunk 4]
```diff
@@ -6585,10 +6585,18 @@ There is a difference in the behavior of the `GetLastWin32Error` method on .NET
         <Parameter Name="ptr" Type="System.IntPtr" Index="0" FrameworkAlternate="netcore-1.1;netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netstandard-2.1;netcore-3.1" />
       </Parameters>
       <Docs>
-        <param name="ptr">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="ptr">The address of the first character of the unmanaged string.</param>
+        <summary>Allocates a managed <see cref="T:System.String" /> and copies all characters up to the first null character from an unmanaged UTF-8 string into it.</summary>
```
Comment (carlossanlop) [inline xml/System.Runtime.InteropServices/Marshal.xml]:
```suggestion
        <summary>Allocates a managed <see cref="T:System.String" /> and copies all characters up to the first <see langword="null" /> character from an unmanaged UTF-8 string into it.</summary>
```

**Comment by AaronRobinsonMSFT** [comment]
@carlossanlop I don't have permission to merge. Feel free to merge whenever is convenient.

**Pull request merged**